staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -510,10 +510,12 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 //
 // It must be called while holding the Authenticator's mutex.
 func (a *Authenticator) wrapCmdRunErrorLocked(err error) error {
-	switch err.(type) {
-	case *exec.Error: // Binary does not exist (see exec.Error).
+	execError := &exec.Error{}
+	exitError := &exec.ExitError{}
+	switch {
+	case errors.As(err, &execError): // Binary does not exist (see exec.Error).
```

Comment:
I couldn't figure out how to work in `os.IsNotExist()` here. I tried both `errors.As(err, &execError) && os.IsNotExist(err)` and `errors.As(err, &execError) && os.IsNotExist(execError)`, but I couldn't make these logical expressions true.